### PR TITLE
[dxa] Add option to dump reflection

### PR DESCRIPTION
**Pull request by python3kgae**
New option -dumpreflection is added to dump reflection data.
This is to support %D3DReflect test in lit.

### CODE REVIEW

**Review by llvm-beanz — COMMENTED** [review]
We need test cases for this. You can add tests to the hcttestcmds tests for Windows and the DXC tests for the lit command line tests on Linux and Windows.

File: tools/clang/tools/dxa/dxa.cpp [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include "dxc/DxilContainer/DxilContainer.h"
 #include "dxc/DxilRootSignature/DxilRootSignature.h"
 #include "dxc/Test/RDATDumper.h"
+#include "dxc/Test/D3DReflectionDumper.h"
```
[the remark this answers is not in the document]
Comment (python3kgae) [reply tools/clang/tools/dxa/dxa.cpp:22]: Moved to DXILContainer.

**Comment by python3kgae** [comment]
> We need test cases for this. You can add tests to the hcttestcmds tests for Windows and the DXC tests for the lit command line tests on Linux and Windows.

Done.

**Review by llvm-beanz — APPROVED** [review]
LGTM

**Pull request merged**